### ISSUE TITLE
[bug] `conan --version` fails to report version on a read-only file system

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#15050]
### Environment details

* Operating System+version: `Linux debian12 6.5.7-x64v3-xanmod1 #0~20231010.gfdab4ec SMP PREEMPT_DYNAMIC Tue Oct 10 21:39:15 UTC x86_64 GNU/Linux`
* Compiler+version: `gcc (Debian 12.2.0-14) 12.2.0`
* Conan version: `Conan version 2.1.0-dev`
* Python version: `Python 3.11.6`

### Steps to reproduce

1. `mkdir ~/.test_conan`
2. `chmod gou-w -R ~/.conan_test`
3. `CONAN_HOME=~/.conan_test conan --version`
4. also, verbose traceback cannot be shown via `CONAN_HOME=~/.conan_test conan --version -vv`

### Logs

```
ERROR: [Errno 13] Permission denied: '/home/sse4/.conan_test/settings.yml'
```

**Comment by SSE4** [comment]
related to earlier reported issue #14732
in general, information-only commands (like `conan --version`, `conan cache path`, `conan profile list`, `conan graph info` and so on) should be working fine on read-only file systems.
especially actual for a cloud environment (e.g. CI systems) where runner user has limited permissions.

**Comment by memsharded** [comment]
Hi @SSE4 

Thanks for your feedback.

All the Conan commands instantiate the API that will read the cache, will run migrations, will load data files from the cache and will **create** them if they are not existing. Even the ``conan --version`` will read the current version, load the ``version.txt`` in the cache (create if not existing), compare and migrate if not matching. This will not change, it would be breaking.

So in short, Conan cannot operate in a folder that is read-only, this is not a bug, this is expected behavior. At most we could discuss if the error message could be more explicit, but I think at this moment the ``Permission denied`` would be enough.

**Comment by memsharded** [comment]
Closing ticket as responded

**Comment by SSE4** [comment]
@memsharded why closed? the question is still not answered, and it's still not clear how to user conan in restricted environment (where you don't have write permissions to file system, e.g. cloud).

**Comment by memsharded** [comment]
The response is there:

> So in short, Conan cannot operate in a folder that is read-only, this is not a bug, this is expected behavior

Conan sometimes need to write files to the cache. If it is a read-only system it will fail, this is expected behavior. Conan needs to be able to write information in the cache, it is its storage, database, configuration, migrations... everything is there. It should be writable in the first place or otherwise the folder would be empty if Conan cannot write there.
